extensions/wikia/ArchiveWikiForum/ArchiveWikiForumHooks.class.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+class ArchiveWikiForumHooks {
+
+    /**
+     * Show the notification informing the user that wiki forums have been disabled
+     * @param Article $article
+     * @param $outputDone
+     * @param $useParserCache
+     * @return bool
+     */
+    public static function onArticleViewHeader( &$article, &$outputDone, &$useParserCache ) {
+        $title = $article->getTitle();
+        if ( static::isForumNS( $title->getNamespace() ) ) {
+            $app = F::app();
+            $html = $app->renderView(
+                ArchiveWikiForumController::class,
+                ArchiveWikiForumController::ARCHIVED_FORUM_NOTIFICATION
+            );
+            $app->wg->Out->addHTML( $html );
+        }
+        return true;
+    }
+
+    /**
+     * Override button on forum
+     * @param WikiaResponse $response
+     * @param $ns
+     * @param $skin
+     * @return bool
+     */
+    public static function onPageHeaderIndexAfterActionButtonPrepared( $response, $ns, $skin ) {
+        $app = F::app();
+        $title = $app->wg->Title;
+        $user = $app->wg->User;
+
+        if ( static::isForumNS( $title->getNamespace() ) && !static::canEditOldForum( $user ) ) {
+            $action = [
+                'class' => '',
+                'text' => wfMessage('viewsource')->escaped(),
+                'href' => $title->getLocalURL( [ 'action' => 'edit' ] ),
+                'id' => 'ca-viewsource',
+                'primary' => 1
+            ];
+            $response->setVal( 'actionImage', MenuButtonController::LOCK_ICON );
+            $response->setVal( 'action', $action );
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @param Title $title
+     * @param User $user
+     * @param $action
+     * @param $result
+     * @return bool
+     */
+    public static function onGetUserPermissionsErrors( Title &$title, User &$user, $action, &$result ) {
+        if ( $action == 'read' ) {
+            return true;
+        }
+
+        if ( static::isForumNS( $title->getNamespace() ) && !static::canEditOldForum( $user ) ) {
+            $result = ['protectedpagetext'];
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @param User $user
+     * @return bool
+     */
+    private static function canEditOldForum( User $user ) {
+        return $user->isAllowed( 'forumoldedit' );
+    }
+
+    private static function isForumNS( $namespace ) {
```

Comment:
So I'm not sure if this affects this at all (especially since it was used in previous code) but there's a method like that in the `WallHelper`:

```
	public static function clearNamespaceList( $namespaces ) {
		if ( !empty( F::app()->wg->EnableForumExt ) ) {
			if ( isset( $namespaces[NS_FORUM] ) ) {
				unset( $namespaces[NS_FORUM] );
			}

			if ( isset( $namespaces[NS_FORUM_TALK] ) ) {
				unset( $namespaces[NS_FORUM_TALK] );
			}

			if ( isset( $namespaces[NS_WIKIA_FORUM_TOPIC_BOARD] ) ) {
				unset( $namespaces[NS_WIKIA_FORUM_TOPIC_BOARD] );
			}
		}

		if ( !empty( F::app()->wg->EnableWallExt ) ) {
			if ( isset( $namespaces[NS_USER_WALL] ) ) {
				unset( $namespaces[NS_USER_WALL] );
			}

			if ( isset( $namespaces[NS_USER_WALL_MESSAGE_GREETING] ) ) {
				unset( $namespaces[NS_USER_WALL_MESSAGE_GREETING] );
			}

			if ( isset( $namespaces[NS_USER_WALL_MESSAGE] ) ) {
				$namespaces[NS_USER_WALL_MESSAGE] = wfMessage( 'wall-recentchanges-wall-thread' )->text();
			}
		}

		return $namespaces;
	}
```